### PR TITLE
Reinvent the wheel [RDY]

### PR DESCRIPTION
**Pull request by mugling**
- Deprecates variable `bigness`
- Specify instead `islot_wheel::diameter` and `islot_wheel::width`
- Steerable variants inherit from their parent types
- No behavioral changes
- Another step closer to #17062

### CODE REVIEW

**Comment by Coolthulhu** [comment]
Trivial conflict with #17167

**Comment by Coolthulhu** [comment]
Note: There is no way to view total wheel size at the moment. This is a significant regression, but it didn't start here.

**Comment by Coolthulhu** [comment]
Crap, I forgot to run `--check-mods`

**Comment by mugling** [comment]
> Crap, I forgot to run --check-mods

Currently broken due to generic guns but I'm guessing blazemod again?

**Comment by Vollinger** [comment]
it sure is, no mo tank threads, no blob-grown parts, no blazemod-specific vehicle armor parts or turrets

`src/veh_type.cpp:533 [static void vehicle_prototype::finalize()] unknown vehicle part tread3 in tank_medium`

File: src/item.cpp [hunk 1]
```diff
@@ -3290,11 +3279,6 @@ bool item::destroyed_at_zero_charges() const
     return (is_ammo() || is_food());
 }
 
-bool item::is_var_veh_part() const
```
Comment (BevapDin) [inline src/item.cpp]: You should also remove the function declaration when you remove a function definition. People might use the declared function and will get obscure linker errors.

**Pull request merged**